### PR TITLE
Pretty Success Message when pod pushed

### PR DESCRIPTION
**Pull request by icanzilb**
Adding pretty output when pushed successfully and not in verbose mode:

image

is it safe to assume there's always at least one message? I'm actually just using it to get the date at trunk.rb: 91 ... now that I think about it - maybe just take the current date and print it in the console? 

Feelings?

### CODE REVIEW

**Comment by orta** [comment]
https please - https://cocoapods.org/pods/ORStackView

**Comment by orta** [comment]
Sometimes we send down messages "Trunk is down", "Maintenance mode" etc - can you make a check for the response being a `400` somewhere before printing success?

**Comment by icanzilb** [comment]
Okay, it looked like all paths just raise and exit, so didn't account for error http responses. I'm on it

**Comment by orta** [comment]
( That could be true, I don't know the codebase too well )

**Comment by icanzilb** [comment]
oh wait, I guess I was wrong - this one does NOT raise 👍 
```
      def request_url(action, url, *args)
        response = create_request(action, url, *args)
        if (400...600).cover?(response.status_code)
          print_error(response.body)
        end
        response
      end
```

File: lib/pod/command/trunk/push.rb [hunk 1]
```diff
@@ -64,14 +64,20 @@ def run
           validate_podspec
           json = push_to_trunk
           update_master_repo
-          print_messages(json['data_url'], json['messages'])
+          
+	      if (400...600).cover?(@push_status)
```
Comment (orta) [inline lib/pod/command/trunk/push.rb]: spacceeess

**Comment by orta** [comment]
I have no idea if those tests pass, I just threw them together in the code editor 👯‍♂️

**Comment by icanzilb** [comment]
ok this may take a while .. can't isntall rspec because needs an older version of json, older version of json can't install because of errors in the code 🤔

**Comment by segiddins** [comment]
@icanzilb let me fix that for you, sorry. Ruby 2.2+ and that version of json don't play nicely

**Comment by icanzilb** [comment]
@segiddins I saw your message

```
# This is the version that ships with OS X 10.10, so be sure we test against it.
# At the same time, the 1.7.7 version won't install cleanly on Ruby > 2.2,
# so we use a fork that makes a trivial change to a macro invocation.
```
and I thought I'd be able to get things run just fine but 🤔

File: lib/pod/command/trunk.rb [hunk 2]
```diff
@@ -70,16 +70,31 @@ def print_error(body)
...
-      def print_messages(data_url, messages)
-        # Using UI.labeled here is dangerous, as it wraps the URL and indents
-        # it, which breaks the URL when you try to copy-paste it.
-        UI.puts "  - Data URL: #{data_url}"
-
-        messages = messages.map do |entry|
-          at, message = entry.to_a.flatten
-          "#{formatted_time(at)}: #{message}"
+      def print_messages(data_url, messages, spec = nil, action = nil)
+        if verbose? || nil?(spec)
```
Comment (segiddins) [inline lib/pod/command/trunk.rb]: I think you mean `spec.nil?` here

File: lib/pod/command/trunk.rb [hunk 3]
```diff
@@ -70,16 +70,31 @@ def print_error(body)
...
+          UI.puts "  - Data URL: #{data_url}"
+
+          messages = messages.map do |entry|
+            at, message = entry.to_a.flatten
+            "#{formatted_time(at)}: #{message}"
+          end
+          UI.labeled 'Log messages', messages
+        else
+          UI.puts
+          UI.puts '-' * 80
```
Comment (segiddins) [inline lib/pod/command/trunk.rb]: please extract the `80` into a variable since its used twice

File: lib/pod/command/trunk.rb [hunk 4]
```diff
@@ -70,16 +70,31 @@ def print_error(body)
...
+          UI.puts " 🎉  Congrats"
+          UI.puts
+          UI.puts " 🚀  #{spec.name} (#{spec.version}) successfully #{action}"
+          if messages.count > 0
```
Comment (segiddins) [inline lib/pod/command/trunk.rb]: `!messages.empty?`

File: lib/pod/command/trunk.rb [hunk 5]
```diff
@@ -70,16 +70,31 @@ def print_error(body)
...
+            at = messages[0].to_a.flatten[0]
```
Comment (segiddins) [inline lib/pod/command/trunk.rb]: `at = messages.first.to_a.flatten.first` ?

File: lib/pod/command/trunk/push.rb [hunk 6]
```diff
@@ -64,14 +64,20 @@ def run
...
+            print_messages(json['data_url'], json['messages'], spec, 'published')
+          end
         end
 
         private
 
         def push_to_trunk
           response = request_path(:post, "pods?allow_warnings=#{@allow_warnings}",
                                   spec.to_json, auth_headers)
+          @push_status = response.status_code
```
Comment (segiddins) [inline lib/pod/command/trunk/push.rb]: maybe return both the push status and the json for the redirected url in this method, instead of setting the ivar?

**Comment by segiddins** [comment]
@icanzilb if you post the output of `bundle env` and the error you're getting when running `bundle install` we might be able to help figure out whats going on

File: lib/pod/command/trunk/push.rb (hunk 6 above)
Comment (icanzilb) [inline lib/pod/command/trunk/push.rb]: got to chek how to do that, but it si probably better yes

**Comment by icanzilb** [comment]
ok, here it goes @segiddins 

`bundle env`

```
Environment

    Bundler   1.13.4
    Rubygems  2.4.8
    Ruby      2.2.0p0 (2014-12-25 revision 49005) [x86_64-darwin16]
    GEM_HOME  /Users/marin/.rvm/gems/ruby-2.2.0
    GEM_PATH  /Users/marin/.rvm/gems/ruby-2.2.0:/Users/marin/.rvm/gems/ruby-2.2.0@global
    RVM       1.27.0 (latest)
    Git       2.8.2
    rubygems-bundler (1.4.4)

Gemfile

    source 'https://rubygems.org'

    gemspec

    # This is the version that ships with OS X 10.10, so be sure we test against it.
    # At the same time, the 1.7.7 version won't install cleanly on Ruby > 2.2,
    # so we use a fork that makes a trivial change to a macro invocation.
    gem 'json', :git => 'https://github.com/segiddins/json.git', :branch => 'seg-1.7.7-ruby-2.2'

    group :development do
      gem 'cocoapods',      :git => "https://github.com/CocoaPods/CocoaPods.git", :branch => 'master'
      gem 'cocoapods-core', :git => "https://github.com/CocoaPods/Core.git", :branch => 'master'
      gem 'claide',         :git => 'https://github.com/CocoaPods/CLAide.git', :branch => 'master'

      gem 'bacon'
      gem 'kicker'
      gem 'mocha'
      gem 'mocha-on-bacon'
      gem 'prettybacon'
      gem 'webmock'

      gem 'codeclimate-test-reporter', :require => nil
      gem 'rubocop'
    end

Gemfile.lock

    GIT
      remote: https://github.com/CocoaPods/CLAide.git
      revision: 8a8f26fb5d6e2b0d604bdb3c2cca14fab4f80adc
      branch: master
      specs:
        claide (1.0.1)

    GIT
      remote: https://github.com/CocoaPods/CocoaPods.git
      revision: 1e3e9ae1cfbed00efe306e9b6f48cca1bfffabca
      branch: master
      specs:
        cocoapods (1.1.0.rc.3)
          activesupport (>= 4.0.2, < 5)
          claide (>= 1.0.1, < 2.0)
          cocoapods-core (= 1.1.0.rc.3)
          cocoapods-deintegrate (>= 1.0.1, < 2.0)
          cocoapods-downloader (>= 1.1.1, < 2.0)
          cocoapods-plugins (>= 1.0.0, < 2.0)
          cocoapods-search (>= 1.0.0, < 2.0)
          cocoapods-stats (>= 1.0.0, < 2.0)
          cocoapods-trunk (= 1.1.0.beta.1)
          cocoapods-try (>= 1.1.0, < 2.0)
          colored (~> 1.2)
          escape (~> 0.0.4)
          fourflusher (~> 2.0)
          gh_inspector (~> 1.0)
          molinillo (~> 0.5.1)
          nap (~> 1.0)
          xcodeproj (>= 1.3.2, < 2.0)

    GIT
      remote: https://github.com/CocoaPods/Core.git
      revision: a886df7735cc928fc69176a0d606e89c8d8cc7df
      branch: master
      specs:
        cocoapods-core (1.1.0.rc.3)
          activesupport (>= 4.0.2, < 5)
          fuzzy_match (~> 2.0.4)
          nap (~> 1.0)

    GIT
      remote: https://github.com/segiddins/json.git
      revision: a9588bc4334c2f5bf985f255b61c05eafdcd8907
      branch: seg-1.7.7-ruby-2.2
      specs:
        json (1.7.7)

    PATH
      remote: .
      specs:
        cocoapods-trunk (1.1.0.beta.1)
          nap (>= 0.8, < 2.0)
          netrc (= 0.7.8)

    GEM
      remote: https://rubygems.org/
      specs:
        activesupport (4.2.7.1)
          i18n (~> 0.7)
          json (~> 1.7, >= 1.7.7)
          minitest (~> 5.1)
          thread_safe (~> 0.3, >= 0.3.4)
          tzinfo (~> 1.1)
        addressable (2.4.0)
        ast (2.3.0)
        bacon (1.2.0)
        cocoapods-deintegrate (1.0.1)
        cocoapods-downloader (1.1.1)
        cocoapods-plugins (1.0.0)
          nap
        cocoapods-search (1.0.0)
        cocoapods-stats (1.0.0)
        cocoapods-try (1.1.0)
        codeclimate-test-reporter (0.6.0)
          simplecov (>= 0.7.1, < 1.0.0)
        colored (1.2)
        crack (0.4.3)
          safe_yaml (~> 1.0.0)
        docile (1.1.5)
        escape (0.0.4)
        ffi (1.9.14)
        fourflusher (2.0.0)
        fuzzy_match (2.0.4)
        gh_inspector (1.0.2)
        hashdiff (0.3.0)
        i18n (0.7.0)
        kicker (3.0.0)
          listen (~> 1.3.0)
          notify (~> 0.5.2)
        listen (1.3.1)
          rb-fsevent (>= 0.9.3)
          rb-inotify (>= 0.9)
          rb-kqueue (>= 0.2)
        metaclass (0.0.4)
        minitest (5.9.1)
        mocha (1.2.0)
          metaclass (~> 0.0.1)
        mocha-on-bacon (0.2.2)
          mocha (>= 0.13.0)
        molinillo (0.5.1)
        multi_json (1.12.1)
        nap (1.1.0)
        netrc (0.7.8)
        notify (0.5.2)
        parser (2.3.1.4)
          ast (~> 2.2)
        powerpack (0.1.1)
        prettybacon (0.0.2)
          bacon (~> 1.2)
        rainbow (2.1.0)
        rake (10.5.0)
        rb-fsevent (0.9.7)
        rb-inotify (0.9.7)
          ffi (>= 0.5.0)
        rb-kqueue (0.2.4)
          ffi (>= 0.5.0)
        rubocop (0.44.1)
          parser (>= 2.3.1.1, < 3.0)
          powerpack (~> 0.1)
          rainbow (>= 1.99.1, < 3.0)
          ruby-progressbar (~> 1.7)
          unicode-display_width (~> 1.0, >= 1.0.1)
        ruby-progressbar (1.8.1)
        safe_yaml (1.0.4)
        simplecov (0.9.2)
          docile (~> 1.1.0)
          multi_json (~> 1.0)
          simplecov-html (~> 0.9.0)
        simplecov-html (0.9.0)
        thread_safe (0.3.5)
        tzinfo (1.2.2)
          thread_safe (~> 0.1)
        unicode-display_width (1.1.1)
        webmock (2.1.0)
          addressable (>= 2.3.6)
          crack (>= 0.3.2)
          hashdiff
        xcodeproj (1.3.2)
          activesupport (>= 3)
          claide (>= 1.0.1, < 2.0)
          colored (~> 1.2)

    PLATFORMS
      ruby

    DEPENDENCIES
      bacon
      bundler (~> 1.3)
      claide!
      cocoapods!
      cocoapods-core!
      cocoapods-trunk!
      codeclimate-test-reporter
      json!
      kicker
      mocha
      mocha-on-bacon
      prettybacon
      rake (~> 10.0)
      rubocop
      webmock

    BUNDLED WITH
       1.11.2

cocoapods-trunk.gemspec

    # coding: utf-8
    lib = File.expand_path('../lib', __FILE__)
    $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
    require 'cocoapods_trunk'

    Gem::Specification.new do |spec|
      spec.name          = "cocoapods-trunk"
      spec.version       = CocoaPodsTrunk::VERSION
      spec.authors       = ["Eloy Durán"]
      spec.email         = ["eloy.de.enige@gmail.com"]
      spec.summary       = "Interact with trunk.cocoapods.org"
      spec.homepage      = "https://github.com/CocoaPods/cocoapods-trunk"
      spec.license       = "MIT"

      spec.files         = `git ls-files`.split($/)
      spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
      spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
      spec.require_paths = ["lib"]

      spec.add_dependency 'nap', '>= 0.8', '< 2.0'
      spec.add_dependency 'netrc', '0.7.8'
      spec.add_development_dependency "bundler", "~> 1.3"
      spec.add_development_dependency "rake", '~> 10.0'

      spec.required_ruby_version = '>= 2.0.0'
    end
```

`bundle install`

```
Resolving dependencies...
Using rake 10.5.0
Using i18n 0.7.0
Using json 1.7.7 from https://github.com/segiddins/json.git (at seg-1.7.7-ruby-2.2@a9588bc)
Using minitest 5.9.1
Using thread_safe 0.3.5
Using addressable 2.4.0
Using ast 2.3.0
Using bacon 1.2.0
Using bundler 1.13.4
Using claide 1.0.1 from https://github.com/CocoaPods/CLAide.git (at master@8a8f26f)
Using fuzzy_match 2.0.4
Using nap 1.1.0
Using cocoapods-deintegrate 1.0.1
Using cocoapods-downloader 1.1.1
Using cocoapods-search 1.0.0
Using cocoapods-stats 1.0.0
Using netrc 0.7.8
Using cocoapods-try 1.1.0
Using colored 1.2
Using escape 0.0.4
Using fourflusher 2.0.0
Using gh_inspector 1.0.2
Using molinillo 0.5.1
Using docile 1.1.5
Using multi_json 1.12.1
Using simplecov-html 0.9.0
Using safe_yaml 1.0.4
Using ffi 1.9.14
Using hashdiff 0.3.0
Using rb-fsevent 0.9.7
Using notify 0.5.2
Using metaclass 0.0.4
Using powerpack 0.1.1
Using rainbow 2.1.0
Using ruby-progressbar 1.8.1
Using unicode-display_width 1.1.1
Using tzinfo 1.2.2
Using parser 2.3.1.4
Using prettybacon 0.0.2
Using cocoapods-plugins 1.0.0
Using cocoapods-trunk 1.1.0.beta.1 from source at `.`
Using simplecov 0.9.2
Using crack 0.4.3
Using rb-inotify 0.9.7
Using rb-kqueue 0.2.4
Using mocha 1.2.0
Using activesupport 4.2.7.1
Using rubocop 0.44.1
Using codeclimate-test-reporter 0.6.0
Using webmock 2.1.0
Using listen 1.3.1
Using mocha-on-bacon 0.2.2
Using cocoapods-core 1.1.0.rc.3 from https://github.com/CocoaPods/Core.git (at master@a886df7)
Using xcodeproj 1.3.2
Using kicker 3.0.0
Using cocoapods 1.1.0.rc.3 from https://github.com/CocoaPods/CocoaPods.git (at master@1e3e9ae)
Bundle complete! 15 Gemfile dependencies, 56 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

and finally `bundle exec rspec spec`

```
/Users/marin/.rvm/gems/ruby-2.2.0/gems/bundler-1.13.4/lib/bundler/rubygems_integration.rb:336:in `block (2 levels) in replace_gem': rspec-core is not part of the bundle. Add it to Gemfile. (Gem::LoadError)
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/rspec:22:in `<main>'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

I  got it so far, but I thought that if I need to start adding stuff to the Gemfile I'm doing something wrong

**Comment by segiddins** [comment]
Oh silly me, I should've paid more attention to the error message! This repo doesn't use rspec, but rather uses bacon. Try `bundle exec bacon spec`, and my apologies if I accidentally said rspec before!

**Comment by icanzilb** [comment]
thanks Sam, ruby testing isn't for the faint hearted apparently 😆 

`bundle exec bacon spec` doesn't do it for me though 😢 

```
.rvm/gems/ruby-2.2.0/gems/bacon-1.2.0/bin/bacon:116:in `load': cannot load such file -- spec (LoadError)
	from /Users/marin/.rvm/gems/ruby-2.2.0/gems/bacon-1.2.0/bin/bacon:116:in `block in <top (required)>'
	from /Users/marin/.rvm/gems/ruby-2.2.0/gems/bacon-1.2.0/bin/bacon:115:in `each'
	from /Users/marin/.rvm/gems/ruby-2.2.0/gems/bacon-1.2.0/bin/bacon:115:in `<top (required)>'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/bacon:23:in `load'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/bacon:23:in `<main>'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/marin/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by segiddins** [comment]
`bundle exec bacon spec/**/*_spec.rb` should do it

**Comment by icanzilb** [comment]
thanks :) I got `bundle exec bacon spec/command/trunk/*_spec.rb` run the tests !

**Comment by orta** [comment]
submitted some tests - https://github.com/icanzilb/cocoapods-trunk/pull/1

**Comment by icanzilb** [comment]
@orta if those tests run at your place, we might be ready with the PR?

**Comment by icanzilb** [comment]
Thanks orta! 💯

**Comment by orta** [comment]
I've just got the CI green in https://github.com/CocoaPods/cocoapods-trunk/pull/78

**Comment by orta** [comment]
ah bah, I might have been lazy on that merge I sent to you, and you might not have got the commit that I ended dup merging into master :-/

**Comment by icanzilb** [comment]
Since #78 is already merged, I'll pull again from master into my branch 👍

**Comment by icanzilb** [comment]
nope all looks good, let's just wait for travis to finish ...

**Comment by orta** [comment]
Alrighty - thanks mr @icanzilb - let's do it

**Comment by icanzilb** [comment]
I think in the end you did more that I did for this PR, but hey - it was fun! 👍💯💖🚢🐐 Thanks

**Comment by orta** [comment]
With a lot of these kind of things, it is easier to edit than to initially create. Lots of people say "you should do x" but actually doing "x" is the tough bit, helping get that into shape is easier, and a better use of our time

**Pull request merged**